### DISCUSSION TITLE
Train on data with more than three channels

### DISCUSSION THREAD

**Discussion by Anneam1213** [ultralytics/ultralytics#3915] [Q&A]
Hello together,
I would like to know if there is a way to train YOLOv8 on an input source that has more than three (R, B, G) channels. Specifically, I would like to train it on a source which has 6 channels per image. I tried it with some .tif files but it did not work.

Alternatively, could you train directly on some numpy arrays with the specified shape (6, x, y) instead?

Any help would be appreciated.

**Comment by glenn-jocher** [comment]
@Anneam1213 hello!

YOLOv8 is primarily designed to work with standard 3-channel RGB images. Training on images with more than three channels, such as your 6-channel images, isn't supported out-of-the-box. This would require significant modifications to the model architecture and data loading pipeline to handle the additional channels.

As for training on numpy arrays, YOLOv8 expects input data in common image formats (like JPEG, PNG, etc.) and does not natively support training directly on numpy arrays. You would need to convert your numpy arrays to a supported format and ensure they are 3-channel images for compatibility with the model.

For custom requirements like yours, you might consider adapting the model and the data pipeline to handle multi-channel inputs, but this goes beyond the scope of the standard YOLOv8 usage and would require a custom implementation.

For further guidance on model customization, you may refer to our documentation at https://docs.ultralytics.com. 😊

Best of luck with your project!
